### PR TITLE
fix(projection): Use emit and not project for column selection

### LINKED ISSUE #365: bug: project rels are not correct
According to the substrait spec, project rels always append, and removal of columns is done through `emit`. Right now, ibis-substrait only keeps the most recently computed expressions around.

We should change the implementation to always keep the columns from the previous relation and, drop them implicitly using `emit`.

### PR DESCRIPTION
**Pull request by gforsyth**
I based this off of the output for similar queries generated by `isthmus`, the `substrait-java` implementation.

I need to figure out how to test this for correctness.

Fixes #365

### CODE REVIEW

File: ibis_substrait/compiler/translate.py [hunk 1]
```diff
@@ -480,6 +482,11 @@ def value_op(
             function_reference=compiler.function_id(expr),
             output_type=translate(expr.type()),
             arguments=[
+                stalg.FunctionArgument(
```
Comment (bkietz) [inline ibis_substrait/compiler/translate.py:485]: This option doesn't apply to all scalar functions, which I think is what's causing [the bulk of the CI failures](https://github.com/ibis-project/ibis-substrait/actions/runs/3268859832/jobs/5375755332#step:6:1203). I'm not sure what the best ibis::way is to detect what enum args are supported by a function

**Review by bkietz — CHANGES_REQUESTED** [review]

File: ibis_substrait/compiler/translate.py (hunk 1 above)
Comment (gforsyth) [reply ibis_substrait/compiler/translate.py:485]: doing this "correctly" is going to have to wait until we're properly parsing the extension yaml files to read in the options (or until the upstream PR making optional parameters _optional_ lands).

For now this is only impacting decompilation, so I'm going to xfail them and circle back.

File: ibis_substrait/compiler/translate.py [hunk 2]
```diff
@@ -474,12 +476,25 @@ def value_op(
     compiler: SubstraitCompiler,
     **kwargs: Any,
 ) -> stalg.Expression:
+    error_args = []
+    # TODO(gforsyth): remove this brittle workaround after extension parsing is ready
+    if (
+        isinstance(op, ops.Binary)
+        and not isinstance(op, ops.Comparison)
+        and isinstance(op.left.type(), (dt.Integer, dt.Floating, dt.Decimal))
+        and isinstance(op.right.type(), (dt.Integer, dt.Floating, dt.Decimal))
```
Comment (bkietz) [inline ibis_substrait/compiler/translate.py:485]:
```suggestion
        isinstance(op, ops.NimericBinary)
```
or [maybe](https://github.com/ibis-project/ibis/blob/4df30e61862d436f5966ca3eb2381e6941d485bc/ibis/expr/operations/numeric.py#L19)
```suggestion
        isinstance(op, (ops.Add, ops.Multiply, *other_error_needing_ops))
```

Comment (gforsyth) [reply ibis_substrait/compiler/translate.py]: I went with `NumericBinary` and also added in a check to make sure we add in the required `rounding` argument for non-integers

Comment (gforsyth) [reply ibis_substrait/compiler/translate.py]: Scratch that -- `NumericBinary` was added after Ibis 3.0.2

File: ibis_substrait/compiler/translate.py (hunk 1 above)
Comment (gforsyth) [reply ibis_substrait/compiler/translate.py]: @jvanstraten -- that is definitely the better target state.  For the moment, I'm after some small degree of interoperability with pyarrow and setting it to `unspecified` doesn't seem to play nice.

Comment (jvanstraten) [reply ibis_substrait/compiler/translate.py]: Huh, based on the current Acero code I thought it should have. Carry on then :)

**Review by jcrist — APPROVED** [review]
With my admittedly limited understanding of substrait, this looks like a sufficient fix to me. :+1:

**Pull request merged**